File: src/javacodes/math/Palindrome.java
package src.javacodes.math;

public class Palindrome {
    public static boolean isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x != 0))
            return false;
        int temp = x, result = 0;
        boolean ans = false;
        while (temp != 0) {//
            result += temp % 10;// 0
            temp /= 10;
            if (result < x)
                result *= 10;
        }
        if (x == result || x == 0)
            ans = true;
        return ans;
    }

    public static void main(String[] args) {
        System.out.println(isPalindrome(121));
    }
}

File: src/python/math/Palindrome.py
class Palindrome:

    @staticmethod
    def isPalindrome(x: int) -> bool:
        if x < 0 or (x % 10 == 0 and x != 0):
            return False

        temp = x
        result = 0
        ans = False

        while temp != 0:
            result += temp % 10
            temp //= 10
            if result < x:
                result *= 10

        if x == result or x == 0:
            ans = True

        return ans



print(Palindrome.isPalindrome(121))
